src/ApplicationInsights.Kubernetes/ContainerIdProviders/MountInfoContainerIdProvider.cs

```diff
@@ -0,0 +1,65 @@
+#nullable enable
+
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.ApplicationInsights.Kubernetes.Debugging;
+
+namespace Microsoft.ApplicationInsights.Kubernetes.ContainerIdProviders;
+
+internal class MountInfoContainerIdProvider : IContainerIdProvider
+{
+    private const string InfoFilePath = "/proc/self/mountinfo";
+    private const string MatchPattern = @"\/docker\/containers\/(.*?)\/";
```

Comment:
Thanks for pointing out. Yes, slash is not required, but it works;
I put it there because it is the lower bar (works in js as well, for example).

1. I don't know there's any negative impact, or is there?
1. Do you think we I shall keep it very cleaning by remove the slashes? I believe remove them works.

Reply:
Thanks for the clarification! 

It is not a huge deal, but I can think of two negatives related to extra backslashes, both related to code readability:
1. They make the regexp harder to read than necessary.
2. They make me (and probably others) wonder why they are there, and whether they are significant.

Reply:
Updated.